pkg/resource/replication_group/custom_update_api_test.go

```diff
@@ -391,6 +392,67 @@ func TestCustomModifyReplicationGroup_NodeGroup_available(t *testing.T) {
 	})
 }
 
+func TestCustomModifyReplicationGroup_ScaleUpAndDown_And_Resharding(t *testing.T)  {
+	assert := assert.New(t)
+	// Setup
+	var mockOutput svcsdk.ListAllowedNodeTypeModificationsOutput
+	testutil.LoadFromFixture(filepath.Join("testdata", "ListAllowedNodeTypeModificationsOutput.json"), &mockOutput)
+
+	// Tests
+	t.Run("ScaleInAndScaleUp=Diff", func(t *testing.T) {
+		desired := provideResource()
+		latest := provideResource()
+		rgId := "RGID"
+		desired.ko.Spec.ReplicationGroupID = &rgId
+		latest.ko.Spec.ReplicationGroupID = &rgId
+
+		mocksdkapi := &mocksvcsdkapi.ElastiCacheAPI{}
+		mocksdkapi.On("ListAllowedNodeTypeModifications", mock.Anything).Return(&mockOutput, nil)
+		rm := provideResourceManagerWithMockSDKAPI(mocksdkapi)
+
+		var delta ackcompare.Delta
+		delta.Add("Spec.CacheNodeType", "cache.t3.small", "cache.m5.large")
+		delta.Add("Spec.NumNodeGroups", 3, 4)
+
+		var ctx context.Context
+		res, err := rm.CustomModifyReplicationGroup(ctx, desired, latest, &delta)
+		assert.Nil(res)
+		assert.Nil(err)
+		assert.NotEmpty(mocksdkapi.Calls)
+		assert.Equal("ListAllowedNodeTypeModifications", mocksdkapi.Calls[0].Method)
```

Comment:
Please add an assert statement to confirm there is no call to `decrease-replica-count` API.

Reply:
Update to assert empty `SDK` calls